storage/src/mmr/iterator.rs

```diff
@@ -109,6 +109,66 @@ pub(crate) fn nodes_needing_parents(peak_iterator: PeakIterator) -> Vec<u64> {
     peaks
 }
 
+/// Returns the position of the oldest provable node in the represented MMR.
+pub(crate) fn oldest_provable_pos(
+    peak_iterator: PeakIterator,
+    oldest_remembered_node_pos: u64,
+) -> u64 {
+    for (peak_pos, height) in peak_iterator {
+        if peak_pos < oldest_remembered_node_pos {
+            continue;
+        }
+        // We have found the tree containing the oldest remembered node. Now we look for the
+        // highest node in this tree whose left-sibling is pruned (if any). The provable nodes
+        // are those that strictly follow this node. If no such node exists, then all existing
+        // nodes are provable
+        let mut two_h = 1 << height;
+        let mut cur_node = peak_pos;
+        while two_h > 1 {
+            let left_pos = cur_node - two_h;
+            let right_pos = left_pos + two_h - 1;
+            if left_pos < oldest_remembered_node_pos {
+                // found pruned left sibling
+                return right_pos + 1;
+            }
+            two_h >>= 1;
+            cur_node = left_pos;
+        }
+        return oldest_remembered_node_pos;
+    }
+    0 // mmr is empty
+}
+
+/// Returns the position of the oldest node whose digest will be required to prove inclusion of
+/// `provable_pos`.
+///
+/// Forgetting this position will render the node with position `provable_pos` unprovable.
+pub(crate) fn oldest_required_proof_pos(peak_iterator: PeakIterator, provable_pos: u64) -> u64 {
+    for (peak_pos, height) in peak_iterator {
+        if peak_pos < provable_pos {
+            continue;
+        }
+        // We have found the tree containing the node we want to guarantee is provable. We
+        // now walk down the path from its root to this node.
+        let iter = PathIterator::new(provable_pos, peak_pos, height);
+        for (parent_pos, sibling_pos) in iter {
+            if parent_pos == provable_pos {
+                // If we hit the node we are trying to prove while walking the path, then no
+                // older nodes are required to prove it.
+                println!("HELLO!!!!");
```

Comment:
We should remove this

Reply:
hehe, you think?  Leftover from my coverage debugging...